src/IonParserTextRaw.ts

```diff
@@ -1063,26 +1128,118 @@ export class ParserTextRaw {
     return s;
   }
 
-  private _skip_triple_quote_gap(ii: number, end: number) : number {
-    ii += 2; // skip the two quotes we peeked ahead to see
-    while (ii<end) {
-      let ch: number = this._in.valueAt(ii);
-      if (IonText.is_whitespace(ch)) {
-        // do nothing
-      }
-      else if (ch == CH_SQ) {
-        ii+= 3;
-        if (ii > end || this._in.valueAt(ii-2) != CH_SQ || this._in.valueAt(ii-1) != CH_SQ) {
-          return ii;
-        }
+  private indexWhiteSpace(indice : number, acceptComments : boolean) : number {
+      let ch : number = this._in.valueAt(indice);
+      if(!acceptComments){
+          for(; is_whitespace(ch); ch = this._in.valueAt(indice++)){}
+      } else {
+          for(;is_whitespace(ch) || ch === CH_FORWARD_SLASH;ch = this._in.valueAt(indice++)){
+              if (ch === CH_FORWARD_SLASH) {
+                  ch = this._in.valueAt(indice++);
+                  switch (ch) {
+                      case CH_FORWARD_SLASH:
+                          indice = this.indexToNewLine(indice);
+                          break;
+                      case CH_AS:
+                          indice = this.indexToCloseComment(indice);
+                          break;
+                      default:
+                          indice--;
+                          break;
+                  }
+              }
+          }
       }
-      else {
-        this._error("unexpected character");
+      return indice;
+  }
+
+  private indexToNewLine(indice : number){
+      let ch : number = this._in.valueAt(indice);
+      while(ch !== EOF && ch !== CH_NL){
+          if (ch === CH_CR) {
+              if (this._in.valueAt(indice + 1) !== CH_NL){
+                  return indice;
+              }
+          }
+          ch = this._in.valueAt(indice++);
       }
-    }
-    return ii;
+      return indice;
+  }
+
+  private indexToCloseComment(indice : number) : number{
+        while(this._in.valueAt(indice) !== CH_AS && this._in.valueAt(indice + 1) !== CH_FORWARD_SLASH){
+            indice++;
+        }
+    return indice;
   }
 
+
+    private _skip_triple_quote_gap(entryIndex: number, end: number, acceptComments: boolean) : number {
+        let tempIndex : number = entryIndex + 3;
+        let ch: number = this._in.valueAt(tempIndex);
+        tempIndex = this.indexWhiteSpace(tempIndex, acceptComments);
+        if (tempIndex + 2 <= end && this.verifyTriple(tempIndex)) {
+            return tempIndex + 4;
```

Comment:
Why don't you `tempIndex += 4;` here and then you can get rid if the `return` on this line and the `else` block after this? It would make the code cleaner to read.